app/(interactive)/4-1/page.tsx

```diff
@@ -0,0 +1,157 @@
+"use client";
+import { useRouter } from "next/navigation";
+import { getStroke } from "perfect-freehand";
+import React, { useContext, useRef } from "react";
+import AppContext from "../../../components/AppContext";
+
+type Props = {};
+
+const Page = (props: Props) => {
+  function getSvgPathFromStroke(stroke: any) {
+    if (!stroke.length) return "";
+    const d = stroke.reduce(
+      (acc: any[], [x0, y0]: any, i: number, arr: string | any[]) => {
+        const [x1, y1] = arr[(i + 1) % arr.length];
+        acc.push(x0, y0, (x0 + x1) / 2, (y0 + y1) / 2);
+        return acc;
+      },
+      ["M", ...stroke[0], "Q"],
+    );
+    d.push("Z");
+    return d.join(" ");
+  }
+
+  const [hasWrite, setWrite] = React.useState<boolean>(false);
+  const [strokes, setStrokes] = React.useState<number[][][]>([]);
+  const currentStroke = useRef<number[][]>([]);
+  const context = useContext(AppContext);
+  const router = useRouter();
+
+  function handlePointerDown(e: React.PointerEvent<SVGSVGElement>) {
+    setWrite(true);
+    e.currentTarget.setPointerCapture(e.pointerId);
+    var frame = document.getElementById("svg");
+    var pos = frame.getBoundingClientRect();
+    currentStroke.current = [
+      [e.pageX - pos.left, e.pageY - pos.top, e.pressure],
+    ];
+  }
+
+  function handlePointerMove(e: React.PointerEvent<SVGSVGElement>) {
+    if (e.buttons !== 1) return;
+    var frame = document.getElementById("svg");
+    var pos = frame.getBoundingClientRect();
+    currentStroke.current.push([
+      e.pageX - pos.left,
+      e.pageY - pos.top,
+      e.pressure,
+    ]);
+    setStrokes([...strokes]);
+  }
+
+  function handlePointerUp() {
+    // new function to handle new independent stroke
+    setStrokes([...strokes, currentStroke.current]);
+    currentStroke.current = [];
+  }
+
+  function clearStrokes() {
+    setWrite(false);
+    setStrokes([]);
+  }
+
+  const renderedStrokes = strokes.map((stroke, index) => {
+    const pathData = getSvgPathFromStroke(
+      getStroke(stroke, {
+        size: 8,
+        thinning: 0.5,
+        smoothing: 0.5,
+        streamline: 0.5,
+      }),
+    );
+    return <path key={index} d={pathData} />;
+  });
+
+  return (
+    <div
+      className="flex h-full w-full flex-col items-center justify-center"
+      onClick={() => {
+        context.setSession(strokes);
+        router.push("/4-2");
```

Comment:
Try using `Link` from `@next/image` instead of `useRouter` if you don't need any logic behind routing.